### PR TITLE
Add changeset-like prompts, changeset demo

### PR DESCRIPTION
**Pull request by natemoo-re**
https://user-images.githubusercontent.com/7118177/220904083-cd999bad-2ace-4880-b9f2-8aeb4653d9db.mp4

### CODE REVIEW

File: packages/prompts/src/index.ts [hunk 1]
```diff
@@ -374,14 +375,140 @@ export const multiselect = <Options extends Option<Value>[], Value extends Primi
...
+							if (groupActive) {
+								return opt(option, selected ? 'group-active-selected' : 'group-active', options);
+							}
+							if (active && selected) {
+								return opt(option, 'active-selected', options);
+							}
+							if (selected) {
+								return opt(option, 'selected', options);
+							}
+							return opt(option, active ? 'active' : 'inactive', options);
```
Comment (natemoo-re) [inline packages/prompts/src/index.ts:476]: This seems like logic that should be moved into `@clack/core` somehow...

File: packages/prompts/src/index.ts [hunk 2]
```diff
@@ -483,7 +610,8 @@ export const group = async <T>(
 	const promptNames = Object.keys(prompts);
 
 	for (const name of promptNames) {
-		const result = await prompts[name as keyof T]({ results }).catch((e) => {
+		const prompt = prompts[name as keyof T];
+		const result = await prompt({ results })?.catch((e) => {
```
Comment (natemoo-re) [inline packages/prompts/src/index.ts:614]: Allows `p.group({ something: () => void })` to be skipped

File: packages/core/src/prompts/group-multiselect.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		}
+	}
+
+	constructor(opts: GroupMultiSelectOptions<T>) {
+		super(opts, false);
+
+		this.options = Object.keys(opts.options).reduce((acc, key) => {
+			acc.push({ value: key, group: true, label: key }, ...opts.options[key].map(opt => ({ ...opt, group: key })));
+			return acc;
+		}, [] as { value: any, [key: string]: any }[]);
```
Comment (ulken) [inline packages/core/src/prompts/group-multiselect.ts:47]: Easier to read, if you ask me.

```suggestion
		this.options = Object.entries(options).flatMap(([key, option]) => [
	{ value: key, group: true, label: key },
	...option.map((opt) => ({ ...opt, group: key })),
])
```

File: packages/core/src/prompts/group-multiselect.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+		this.on('cursor', (key) => {
+			switch (key) {
+				case 'left':
+				case 'up':
+					this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1;
+					break;
+				case 'down':
+				case 'right':
+					this.cursor = this.cursor === this.options.length - 1 ? 0 : this.cursor + 1;
```
Comment (ulken) [inline packages/core/src/prompts/group-multiselect.ts:62]:
```suggestion
					this.cursor = (this.cursor + 1) % this.options.length;
```

File: packages/core/src/prompts/group-multiselect.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+		this.cursor = Math.max(
+			this.options.findIndex(({ value }) => value === opts.cursorAt),
+			0
+		);
+
+		this.on('cursor', (key) => {
+			switch (key) {
+				case 'left':
+				case 'up':
+					this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1;
```
Comment (ulken) [inline packages/core/src/prompts/group-multiselect.ts:58]:
```suggestion
					this.cursor = (this.options.length + (cursor - 1)) % this.options.length;
```

File: packages/prompts/src/index.ts [hunk 6]
```diff
@@ -374,14 +375,140 @@ export const multiselect = <Options extends Option<Value>[], Value extends Primi
...
+	const opt = (
+		option: Options[number],
+		state: 'inactive' | 'active' | 'selected' | 'active-selected' | 'group-active' | 'group-active-selected' | 'submitted' | 'cancelled',
+		options: Options = [] as any,
+	) => {
+		const label = option.label ?? String(option.value);
+		const isItem = typeof option.group === 'string';
+		const next = isItem && (options[options.indexOf(option) + 1] ?? { group: true });
+		const isLast = isItem && next.group === true;
+		const prefix = typeof option.group === 'string' ? `${isLast ? S_BAR_END : S_BAR} ` : '';
```
Comment (ulken) [inline packages/prompts/src/index.ts:397]:
```suggestion
		const prefix = isItem ? `${isLast ? S_BAR_END : S_BAR} ` : '';
```

File: packages/prompts/src/index.ts [hunk 7]
```diff
@@ -374,14 +375,140 @@ export const multiselect = <Options extends Option<Value>[], Value extends Primi
...
+							return opt(option, active ? 'active' : 'inactive', options);
+						})
+						.join(`\n${color.yellow(S_BAR)}  `)}\n${footer}\n`;
+				}
+				default: {
+					return `${title}${color.cyan(S_BAR)}  ${this.options
+						.map((option, i, options) => {
+							const selected = this.value.includes(option.value) || (option.group === true && this.isGroupSelected(option.value));
+							const active = i === this.cursor;
+							const groupActive = !active &&  typeof option.group === 'string' && this.options[this.cursor].value === option.group;
```
Comment (ulken) [inline packages/prompts/src/index.ts:485]:
```suggestion
							const groupActive = !active && typeof option.group === 'string' && this.options[this.cursor].value === option.group;
```

**Review by ulken — CHANGES_REQUESTED** [review]
Super slick demo!

Do you plan on adding group versions of any other prompts?

File: packages/core/src/prompts/group-multiselect.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+			if (this.isGroupSelected(group)) {
+				this.value = this.value.filter((v: string) => groupedItems.findIndex(i => i.value === v) === -1);
+			} else {
+				this.value = [...this.value, ...groupedItems.map(v => v.value)];
+			}
+			this.value = Array.from(new Set(this.value));
+		} else {
+			const selected = this.value.includes(item.value);
+			this.value = selected
+				? this.value.filter((value: T['value']) => value !== item.value)
```
[the remark this answers is not in the document]
Comment (natemoo-re) [reply packages/core/src/prompts/group-multiselect.ts:36]: Ha, fair enough!

File: packages/core/src/prompts/group-multiselect.ts (hunk 3 above)
Comment (natemoo-re) [reply packages/core/src/prompts/group-multiselect.ts:47]: Yep, great call. I never like using `reduce`

File: packages/core/src/prompts/group-multiselect.ts (hunk 5 above)
Comment (natemoo-re) [reply packages/core/src/prompts/group-multiselect.ts:58]: This feels a little too clever for me, to be honest.

File: packages/prompts/src/index.ts (hunk 1 above)
Comment (natemoo-re) [reply packages/prompts/src/index.ts:476]: Yeah I don't know what the API should look like, but the fact that we have to track state in the rendering logic feels leaky. That's the whole point of `@clack/core`!

File: packages/prompts/src/index.ts (hunk 2 above)
Comment (natemoo-re) [reply packages/prompts/src/index.ts:614]: Yep, this allows you to conditionally skip by returning `undefined`.

Agreed on the weird `catch => throw`, will have to think about that some more.

File: packages/prompts/src/index.ts (hunk 7 above)
Comment (ulken) [reply packages/prompts/src/index.ts]: Just to clarify: this suggestion removes an extra whitespace before `typeof`. Still seem to be present? Don't care?

**Review by ulken — APPROVED** [review]

**Pull request merged**